helper/schema/grpc_provider.go

```diff
@@ -1544,6 +1563,24 @@ func (s *GRPCProviderServer) ApplyResourceChange(ctx context.Context, req *tfpro
 			return resp, nil
 		}
 
+		if !res.ResourceBehavior.AllowNullIdentity {
+			isFullyNull := true
+			for _, v := range newIdentityVal.AsValueMap() {
+				if !v.IsNull() {
+					isFullyNull = false
+					break
+				}
+			}
+
+			if isFullyNull {
+				resp.Diagnostics = convert.AppendProtoDiag(ctx, resp.Diagnostics, fmt.Errorf(
+					"Missing Resource Identity After Create: The Terraform provider unexpectedly returned no resource identity after having no errors in the resource create. "+
+						"This is always a problem with the provider and should be reported to the provider developer",
+				))
```

Comment:
nit on message: This isn't an error that can only happen during create right? Technically this logic is run during any call to `ApplyResoureChange`, which could be a `create`, `update`, or `delete/destroy`